pkg/unshare/unshare_linux.go

```diff
@@ -464,11 +464,6 @@ func bailOnError(err error, format string, a ...interface{}) { // nolint: golint
 
 // MaybeReexecUsingUserNamespace re-exec the process in a new namespace
 func MaybeReexecUsingUserNamespace(evenForRoot bool) {
```

Comment:
If this changes the semantics of the function in caller-visible ways, I’d expect that to show up _at least_ in the documentation, if not in the name.

IMHO doc strings should be enough for a caller to understand what the function does (from the caller’s perspective) without needing to read the implementation, and this one is way below that standard.